### PR TITLE
FEAT(client): Implement a mute cue

### PR DESCRIPTION
**Pull request by pkern**
The mute cue is played when speech is detected while the local client is
muted. This reminds the user that they are still muted, which is
especially useful when the overlay/client indication is not available
(e.g. in full-screen games).

This currently reuses the existing "audio off" cue.

Implements #4866

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]
I'll review the actual changes in the coming days but for now could you separate the translation changes into their own commit, please?

You can achieve that by using`git reset HEAD~1` then `git restore src/mumble/*.ts` and then commiting your code changes again.
After that, please run `scripts/updatetranslations.py` which will create a new commit for the translation changes. And finally push both of these commits to your branch (force-push).

Having the translation changes separate facilitates reviewing :)

**Comment by pkern** [comment]
> I'll review the actual changes in the coming days but for now could you separate the translation changes into their own commit, please?
> Having the translation changes separate facilitates reviewing :)

Done. I mostly did it because the checks were yelling at me, but sure, I split them out. ;-)

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [inline src/mumble/AudioInput.cpp:947]: I think I would turn this into a static member variable in the `AudioOutput` class. Also I would suggest to use `constexpr` instead of `const` for this kind of stuff.

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [inline src/mumble/AudioInput.cpp:1083]: This could be shortened as
```suggestion
		bTalkingWhenMuted = bIsSpeech;
```
The `if` makes the semantics a bit clearer though, so in the end I think I don't mind which version is used (I assume the extra branch will be optimized away anyways). Your call

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [inline src/mumble/AudioInput.cpp:9]: What is this needed for? :eyes:

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [inline src/mumble/AudioInput.cpp:1119]:
```suggestion
		if (ao) {
		    if (Global::get().s.bTxAudioCue) {
			      if (bIsSpeech && !bPreviousVoice) {
				      ao->playSample(Global::get().s.qsTxAudioCueOn);
			      } else if (!bIsSpeech && bPreviousVoice) {
				      ao->playSample(Global::get().s.qsTxAudioCueOff);
			      }
		    }
		    
		    if (Global::get().s.bTxMuteCue && bTalkingWhenMuted) {
			      if (!qetLastMuteCue.isValid() || qetLastMuteCue.elapsed() > MUTE_CUE_DELAY) {
				      qetLastMuteCue.start();
				      ao->playSample(Global::get().s.qsTxMuteCue);
			      }
		    }
		}
```

File: src/mumble/AudioInput.h
Comment (Krzmbrzl) [inline src/mumble/AudioInput.h:9]:
```suggestion
#include QElapsedTimer>
```
This is actually the preferred way of including Qt headers. We only found out about that very recently and therefore the rest of the code base partially does use the other style.

File: src/mumble/Settings.cpp
Comment (Krzmbrzl) [inline src/mumble/Settings.cpp:337]:
```suggestion
	bTxMuteCue  = true;
```
I think we can default to enabling this feature as it seems very handy. Therefore I think most folks like being warned about speaking when being muted and the ones that don't, can then go and turn it off for themselves.

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: src/mumble/AudioInput.cpp
Comment (pkern) [reply src/mumble/AudioInput.cpp]: I just found what is arguably a bug: If we are deafened, the mute cue should not play. So I added that to the expression, PTAL.

File: src/mumble/Settings.cpp
Comment (pkern) [reply src/mumble/Settings.cpp]: I did that, thanks. One case I could imagine where it's annoying is if you are passively listening (e.g. on the c3 lingo server). However in interactive cases it seems more useful. I also guess we don't have any infrastructure to easily toggle the default later, right? Then maybe it's better to start out this way and maybe offer a button in the top bar if that's something that needs toggling more often?

File: src/mumble/AudioInput.cpp
Comment (pkern) [reply src/mumble/AudioInput.cpp]: Good point, done.

File: src/mumble/AudioInput.cpp
Comment (pkern) [reply src/mumble/AudioInput.cpp]: Removed. I had an incomplete type in an intermediate state where I tried to use slots.

**Review by pkern — COMMENTED** [review]
PTAL. I kept the review updates in a separate commit, let me know if I should merge it into the main one.

**Comment by pkern** [comment]
I guess the commit-check is clear enough and this requires force-pushing of a squashed version.

**Comment by pkern** [comment]
Rebased onto the format change. I also changed from `on.ogg` to `off.ogg`. I don't personally understand the value of the audio cue yet, but it probably makes sense when both are enabled to play the off sound rather than the "I'm now sending" cue.

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [reply src/mumble/AudioInput.cpp]: Yeah sorry, I meant `AudioInput`

File: src/mumble/AudioInput.h
Comment (Krzmbrzl) [reply src/mumble/AudioInput.h]: If clang-format is happy with it, we are too :)
Did you run clang-format yet? If not, pull the latest changes and then just run it on all files that you have edited. That should take care of all formatting aspects.

File: src/mumble/Settings.cpp
Comment (Krzmbrzl) [reply src/mumble/Settings.cpp]: Wdym "changing it later"? We can change it in any release if we wish to do so.

> maybe offer a button in the top bar if that's something that needs toggling more often?

Yeah that sounds good

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [inline src/mumble/AudioInput.cpp:1086]: 1. What you want to check here is `Global::get().s.bDeaf` - that's where the local user's deaf status is (somewhat) reliably stored.
2. I think semantically the check for whether the user is deafened should not be done here as the user might still be talking while being muted. _But_ in that case we don't need to play the cue (though if we did there would probably be no difference since the user won't hear it anyways ^^). In short: I would check for this further down when you actually decide whether or not to play the cue.

File: src/mumble/AudioInput.cpp
Comment (Krzmbrzl) [inline src/mumble/AudioInput.cpp:1115]:
```suggestion
				if (bIsSpeech && !bPreviousVoice) {
					ao->playSample(Global::get().s.qsTxAudioCueOn);
				} else if (!bIsSpeech && bPreviousVoice) {
					ao->playSample(Global::get().s.qsTxAudioCueOff);
				}
```
When we're changing these already, we might as well add these curly brackets (we prefer always having them)

File: src/mumble/AudioInput.h
Comment (Krzmbrzl) [inline src/mumble/AudioInput.h:233]:
```suggestion
	/// The minimum time in ms that has to pass between the playback of two consecutive mute cues.
	static constexpr unsigned int iMuteCueDelay = 5000;
```

File: src/mumble/AudioInput.cpp
Comment (pkern) [reply src/mumble/AudioInput.cpp]: Ack, implemented both. Thank you!

File: src/mumble/AudioInput.cpp
Comment (pkern) [reply src/mumble/AudioInput.cpp]: Done.

File: src/mumble/AudioInput.h
Comment (pkern) [reply src/mumble/AudioInput.h]: Done, thanks!

File: src/mumble/Settings.cpp
Comment (pkern) [reply src/mumble/Settings.cpp]: I guess I sort of assumed that any original default would be written into the local user config. If we only persist differences, that'd be easy. ;-)

Do we need the button in this rev of the change or can we wait for beta feedback?

File: src/mumble/AudioInput.h
Comment (pkern) [reply src/mumble/AudioInput.h]: Ran clang-format now. Nice stuff.

**Review by Krzmbrzl — APPROVED** [review]

**Comment by Krzmbrzl** [comment]
Thank you very much for your contribution :+1:

**Comment by lePoMo** [comment]
Is there a config for this ? I updated my mumble client (Mumble 1.4.230 released 2022-01) and cannot find where to disable this behavior.

**Comment by Krzmbrzl** [comment]
Yes there is a setting under `Audio Input` (at the bottom right under the setting for `Audio cue`)

**Comment by lePoMo** [comment]
Thank you. Not sure how I didn't see that. Also I guess was expecting it in "Messages", but I see how it is different to other things in messages.

**Pull request merged**
